megamek/src/megamek/common/Compute.java

```diff
@@ -1859,6 +1858,93 @@ public static Entity exposed_findC3Spotter(Game game, Entity attacker,
         return findC3Spotter(game, attacker, target);
     }
 
+    /**
+     * Function that attempts to find one TAG spotter, or the best TAG spotter,
+     * @param game
+     * @param attacker should be an artillery unit
+     * @param target (if null, do not return a spotter!)
+     * @param stopAtFirst if we only want to know that there is one, not which is best
+     * @return The Spotter.
+     */
+    public static Entity findTAGSpotter(Game game, Entity attacker, Targetable target, boolean stopAtFirst) {
```

Comment:
When you're calculating movement or firing phase, TAG attacks for the round have already been declared and processed, so I don't really see why there's a need for this expensive calculation. You should be able to go to the "game" object and look through the list of active TAG attacks instead.